src/snowflake/snowpark/udaf.py

```diff
@@ -413,43 +416,45 @@ def register(
             - :func:`~snowflake.snowpark.functions.udaf`
             - :meth:`register_from_file`
         """
-        if not isinstance(handler, type):
-            raise TypeError(
-                f"Invalid handler: expecting a class type, but get {type(handler)}"
-            )
-
-        check_register_args(
-            TempObjectType.AGGREGATE_FUNCTION,
-            name,
-            is_permanent,
-            stage_location,
-            parallel,
-        )
+        parameters = {"handler": handler, "name": name}
```

Comment:
Can use named arguments, I think it is better than creating a new dict and search in it.